core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -372,10 +373,15 @@ private void handleResponse(RetryConfig<HttpResponse> retryConfig, ClientRequest
                                 @Nullable AggregatedHttpResponse aggregatedRes) {
         assert response != null || aggregatedRes != null;
         final RequestLogProperty logProperty =
-                retryConfig.requiresResponseTrailers() ?
-                RequestLogProperty.RESPONSE_TRAILERS : RequestLogProperty.RESPONSE_HEADERS;
+                retryConfig.requiresResponseTrailers() ? null : RequestLogProperty.RESPONSE_HEADERS;
 
-        derivedCtx.log().whenAvailable(logProperty).thenAccept(log -> {
+        final CompletableFuture<RequestLog> logFuture;
+        if (logProperty == null) {
+            logFuture = derivedCtx.log().whenComplete();
```

Comment:
`GrpcClientRetryTest#childrenContextsHaveSameRpcRequest` unittest failed because `RequestLogProperty.RESPONSE_CONTENT` wasn't set, and the chain of this `whenComplete` couldn't be invoked.

Reply:
I'll take some time to think about how I can address this issue.